### PR TITLE
830 update save content

### PR DESCRIPTION
**Pull request by goodviber**
### Context
Some of our buttons and success messages don't match the interaction. We need to update the content so it matches what happens. See below:

All changes made in the About your organisation section
Course status: Published
Button: 'Save and publish'
Success message: 'Your changes have been published'
Inset text above the button: "Changes will appear on Find postgraduate teacher training within 15 minutes."
Remove inset text
Editing an existing location in the Locations section
Course status: Published
Button: 'Save'
Success message: 'Your changes have been published'
Change button to: Save and publish
All changes made in the Requirements and eligibility section
Course status: Published
Button: 'Save'
Success message: 'Your changes have been saved'
Change button to: Save and publish
Change success message to: Your changes have been published
All changes made in the Basic details section
Course status: Published
Button: 'Save'
Success message: 'Your changes have been saved'
Inset text above the button: "Changes will appear on Find postgraduate teacher training within 15 minutes."
Remove inset text
Change button to: Save and publish
Change success message to: Your changes have been published

Plus google doc: https://docs.google.com/document/d/1wp2Js5W1SH3a5XFZD4deZkCr8eYHjwH1ckaHmJp-4pI/edit#
### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -54,6 +54,7 @@
         </div>
       </main>
     </div>
+    <%= debug params %>
```
Comment (goodviber) [inline app/views/layouts/application.html.erb:57]: remember to remove this

**Comment by defong** [comment]
Can we have some screensnots of the items in concern

**Comment by darokel** [comment]
> Can we have some screensnots of the items in concern

Review app should be fine? Might be worth getting @Kateharries to product review

**Comment by darokel** [comment]
Might be worth getting @Kateharries to product review

File: app/controllers/publish/courses/age_range_controller.rb [hunk 2]
```diff
@@ -14,7 +14,7 @@ def edit
 
       def update
         if form_object.valid?
-          flash[:success] = I18n.t("success.saved")
+          flash[:success] = @course.is_published? ? I18n.t("success.value_published", value: "age range") : I18n.t("success.value_saved", value: "age range")
```
Comment (darokel) [inline app/controllers/publish/courses/age_range_controller.rb:17]: This is used in a few places, is it worth putting into a method/concern which other controllers can re-use? then each one would just call the method with the value?

Comment (goodviber) [reply app/controllers/publish/courses/age_range_controller.rb]: Done

File: app/controllers/concerns/success_message.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+module SuccessMessage
+  extend ActiveSupport::Concern
+
+  def details_flash_message(value)
```
Comment (darokel) [inline app/controllers/concerns/success_message.rb:4]: This is much better. I think we should make the method name a bit clearer so whoever uses it knows they're showing a **success** message. Maybe something like `publishable_flash_message` or something 🤷‍♂️

**Comment by darokel** [comment]
I've just left one last comment then we can get this over the line

**Review by darokel — APPROVED** [review]

**Pull request merged**